sensor/kubernetes/certrefresh/certificates/requester.go

```diff
@@ -187,37 +190,45 @@ func (r *genericRequester[ReqT, RespT]) RequestCertificates(ctx context.Context)
 		}
 	}
 
+	// create a new channel for this specific request, and store it in the requests map
 	requestID := uuid.NewV4().String()
-	receiveC := make(chan RespT, 1)
-	r.requests.Store(requestID, receiveC)
-	defer r.requests.Delete(requestID)
+	certsResponseC := make(chan RespT, 1)
```

Comment:
1. It is buffered because that's how the original author intended to prevent deadlocks, see [this comment](https://github.com/stackrox/stackrox/pull/13369/files#diff-293bd4a7ebbe7421c0a1a83a90a3d0c5c9028740f193d23b3d5f1030e0c1f1faR173)

I'll try to refactor this so that this is no longer needed.

2. It's closed [here](https://github.com/stackrox/stackrox/pull/13369/files#diff-293bd4a7ebbe7421c0a1a83a90a3d0c5c9028740f193d23b3d5f1030e0c1f1faR177)

Reply:
Yes, this is one of the reasons why buffered channels are introduced. Unfortunately that only masks the problem and this is something that I would like to avoid.

I am afraid that the links are not leading to the answers - could you check maybe?

Reply:
For context, this is when the buffer was introduced: https://github.com/stackrox/stackrox/pull/463#discussion_r794378441

Reply:
I removed these channels.